libdevice/cmake/modules/SYCLLibdevice.cmake

```diff
@@ -22,7 +22,7 @@ set(install_dest_obj lib${LLVM_LIBDIR_SUFFIX})
 set(install_dest_obj-new-offload lib${LLVM_LIBDIR_SUFFIX})
 set(install_dest_bc lib${LLVM_LIBDIR_SUFFIX})
 
-set(clang $<TARGET_FILE:clang>)
+set(clangxx $<TARGET_FILE:clang> --driver-mode=g++)
```

Comment:
I think the right place to add compiler options is to extend `compile_opts` variable. If I understand the problem correctly, the option we need to set is rather `-x c++`.

Reply:
> If I understand the problem correctly, the option we need to set is rather -x c++

That's not case, `-x c++` is already implicitly set based on the file extension. ICX is specifically looking for the name it is invoked as (`argv[0]`), which set its driver mode.
This is the code where the warning is emitted for folks who have access to it: [clang/lib/Driver.cpp:4098](https://github.com/intel-restricted/applications.compilers.llvm-project/blob/387f14a9e664b1a23dc4cedff568ac93f01ad6e5/clang/lib/Driver/Driver.cpp#L4098-L4103)

Since `clang++` is not a full CMake target (but a symlink) we can't do `$<TARGET_FILE:clang++>`, but the "driver mode" can also be set explicitly via the option as I did here. This is also why I included the option as part of the variable for the the tool "executable"; In my thinking we're invoking `clang++`, but can't spell it exactly as such.

Reply:
Ping @bader @aelovikov-intel

Reply:
I think this falls more to the compilation/offload tools or driver than to SYCL RT. 

@mdtoguchi , @AlexeySachkov , @asudarsa , what are your thoughts on this?

Reply:
I somewhat agree and in-fact my initial reaction was fixing icx to not issue this warning for this case. We're passing it `-fsycl-device` so the host c++ runtime libs are irrelevant. I will revisit that patch, but I believe its not completely wrong to suggest that clang++ be used for C++ sources, so we might apply this too.

Reply:
The use of `clang --driver-mode=g++` does tell the driver to work as the C++ compiler but from what I understand, the diagnostic that is emitted is happening when we are using `-fsycl-device-only`.  The diagnostic is a link specific diagnostic and since we aren't really doing a host link step with `-fsycl-device-only` we can probably address the diagnostic emission with a modification to the compiler driver to be a bit smarter when emitting this diagnostic.